pkg/hardwareutils/bmc/access.go

```diff
@@ -150,3 +157,23 @@ func NewAccessDetails(address string, disableCertificateVerification bool) (Acce
 
 	return factory(parsedURL, disableCertificateVerification)
 }
+
+func checkDNSValid(address string) error {
+
+	// Allowing empty BMC address
+	if address == "" {
+		return nil
+	}
+
+	// Check if its a IPv6/IPv4 address
+	if net.ParseIP(address) != nil {
```

Comment:
In this line, we have a DNS value, why we are parsing assuming that it is IP?

Reply:
In lot of tests we have IPv6 address ( and I think the BMC address hostname can be an IPv6 address) - this check is validate that. IPv4 addresses are anyways DNS standard compliant, so it doesn't need a check - I will remove IPv4 from the comment to avoid this confusion.